mzLib/FlashLFQ/FlashLfqEngine.cs

```diff
@@ -199,14 +219,17 @@ public FlashLfqResults Run()
         {
             _globalStopwatch.Start();
             _ms1Scans = new Dictionary<SpectraFileInfo, Ms1ScanInfo[]>();
-            _results = new FlashLfqResults(_spectraFileInfo, _allIdentifications, MbrDetectionQValueThreshold, PeptideModifiedSequencesToQuantify);
-
+            _results = new FlashLfqResults(_spectraFileInfo, _allIdentifications, MbrDetectionQValueThreshold, PeptideModifiedSequencesToQuantify, IsoTracker);
             // build m/z index keys
             CalculateTheoreticalIsotopeDistributions();
-
             // quantify each file
             foreach (var spectraFile in _spectraFileInfo)
             {
+                // create a new dictionary for isobaric case
+                PeakIndexingEngine indexEngine = new PeakIndexingEngine();
+                indexEngine.IndexMassSpectralPeaks(spectraFile, Silent, _ms1Scans);
```

Comment:
You're indexing the peaks twice for every file.

Also, in the dictionary, you need to serialize and clear the index after you're done analyzing that file.